diff --git a/app/Makefile.am b/app/Makefile.am
index 0b11552..58b4d76 100644
--- a/app/Makefile.am
+++ b/app/Makefile.am
@@ -21,7 +21,7 @@ seafile_SOURCES = seafile.c
 seafile_LDADD =  @CCNET_LIBS@ \
 	$(top_builddir)/lib/libseafile.la \
 	$(top_builddir)/lib/libseafile_common.la \
-	@SEARPC_LIBS@ \
+	@SEARPC_LIBS@ @JANSSON_LIBS@ \
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 @ZLIB_LIBS@
 
 seafile_LDFALGS = @STATIC_COMPILE@ @CONSOLE@
@@ -29,7 +29,7 @@ seafile_LDFALGS = @STATIC_COMPILE@ @CONSOLE@
 if COMPILE_SERVER
 seafserv_tool_SOURCES = seafserv-tool.c
 seafserv_tool_LDADD =  @CCNET_LIBS@ \
-	$(top_builddir)/lib/libseafile.la @SEARPC_LIBS@ \
+	$(top_builddir)/lib/libseafile.la @SEARPC_LIBS@ @JANSSON_LIBS@ \
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 @ZLIB_LIBS@
 
 seafserv_tool_LDFLAGS = @STATIC_COMPILE@ @CONSOLE@ @SERVER_PKG_RPATH@
diff --git a/configure.ac b/configure.ac
index d778274..98c9668 100644
--- a/configure.ac
+++ b/configure.ac
@@ -237,6 +237,7 @@ LIBEVENT_REQUIRED=2.0
 GLIB_REQUIRED=2.16.0
 CCNET_REQUIRED=0.9.3
 SEARPC_REQUIRED=1.0
+JANSSON_REQUIRED=2.2.1
 ZDB_REQUIRED=2.10
 #LIBNAUTILUS_EXTENSION_REQUIRED=2.30.1
 CURL_REQUIRED=7.17
@@ -259,6 +260,10 @@ PKG_CHECK_MODULES(SEARPC, [libsearpc >= $SEARPC_REQUIRED])
 AC_SUBST(SEARPC_CFLAGS)
 AC_SUBST(SEARPC_LIBS)
 
+PKG_CHECK_MODULES(JANSSON, [jansson >= $JANSSON_REQUIRED])
+AC_SUBST(JANSSON_CFLAGS)
+AC_SUBST(JANSSON_LIBS)
+
 PKG_CHECK_MODULES(LIBEVENT, [libevent >= $LIBEVENT_REQUIRED])
 AC_SUBST(LIBEVENT_CFLAGS)
 AC_SUBST(LIBEVENT_LIBS)
diff --git a/controller/Makefile.am b/controller/Makefile.am
index a255d9e..4b9843a 100644
--- a/controller/Makefile.am
+++ b/controller/Makefile.am
@@ -18,6 +18,6 @@ seafile_controller_SOURCES = seafile-controller.c ../common/log.c
 seafile_controller_LDADD = @CCNET_LIBS@ \
 	$(top_builddir)/lib/libseafile_common.la \
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -levent \
-	@SEARPC_LIBS@ @ZLIB_LIBS@
+	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZLIB_LIBS@
 
 seafile_controller_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index 473ad84..c9e3363 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -128,7 +128,7 @@ seaf_daemon_LDADD = $(top_builddir)/lib/libseafile_common.la \
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 -levent \
 	$(top_builddir)/common/cdc/libcdc.la \
 	$(top_builddir)/common/index/libindex.la ${LIB_WS32} \
-	@SEARPC_LIBS@ @CCNET_LIBS@ @GNOME_KEYRING_LIBS@ @LIB_MAC@ @ZLIB_LIBS@
+	@SEARPC_LIBS@ @CCNET_LIBS@ @GNOME_KEYRING_LIBS@ @JANSSON_LIBS@ @LIB_MAC@ @ZLIB_LIBS@
 
 seaf_daemon_LDFLAGS = @STATIC_COMPILE@ @CONSOLE@
 
diff --git a/fuse/Makefile.am b/fuse/Makefile.am
index c0716b1..4de4482 100644
--- a/fuse/Makefile.am
+++ b/fuse/Makefile.am
@@ -43,6 +43,6 @@ seaf_fuse_LDADD = @CCNET_LIBS@ \
 				  @GLIB2_LIBS@ @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ \
                   -lsqlite3 -levent \
 				  $(top_builddir)/common/cdc/libcdc.la \
-				  @SEARPC_LIBS@ @ZDB_LIBS@ @FUSE_LIBS@ @ZLIB_LIBS@
+				  @SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @FUSE_LIBS@ @ZLIB_LIBS@
 
 seaf_fuse_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
diff --git a/httpserver/Makefile.am b/httpserver/Makefile.am
index e2805a6..2668fba 100644
--- a/httpserver/Makefile.am
+++ b/httpserver/Makefile.am
@@ -50,7 +50,7 @@ httpserver_LDADD = -levent -levhtp -lssl -levent_openssl \
 	@CCNET_LIBS@ \
 	$(top_builddir)/lib/libseafile.la \
 	$(top_builddir)/common/cdc/libcdc.la \
-	@SEARPC_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ \
+	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ \
 	@LIBARCHIVE_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
 
 httpserver_LDFLAGS = @STATIC_COMPILE@
diff --git a/server/Makefile.am b/server/Makefile.am
index 36da709..fb1740b 100644
--- a/server/Makefile.am
+++ b/server/Makefile.am
@@ -102,6 +102,6 @@ seaf_server_LDADD = @CCNET_LIBS@ \
 	$(top_builddir)/common/index/libindex.la \
 	@GLIB2_LIBS@ @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 -levent \
 	$(top_builddir)/common/cdc/libcdc.la \
-	@SEARPC_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
+	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
 
 seaf_server_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
diff --git a/server/gc/Makefile.am b/server/gc/Makefile.am
index ff2e4d0..740ad52 100644
--- a/server/gc/Makefile.am
+++ b/server/gc/Makefile.am
@@ -51,7 +51,7 @@ seafserv_gc_LDADD = @CCNET_LIBS@ \
 	$(top_builddir)/common/cdc/libcdc.la \
 	$(top_builddir)/lib/libseafile_common.la \
 	@GLIB2_LIBS@ @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 -levent \
-	@SEARPC_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
+	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
 
 seafserv_gc_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
 
@@ -64,7 +64,7 @@ seaf_fsck_LDADD = @CCNET_LIBS@ \
 	$(top_builddir)/common/cdc/libcdc.la \
 	$(top_builddir)/lib/libseafile_common.la \
 	@GLIB2_LIBS@ @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 -levent \
-	@SEARPC_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
+	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
 
 seaf_fsck_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
 
@@ -76,7 +76,7 @@ seaf_migrate_LDADD = @CCNET_LIBS@ \
 	$(top_builddir)/common/cdc/libcdc.la \
 	$(top_builddir)/lib/libseafile_common.la \
 	@GLIB2_LIBS@ @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 -levent \
-	@SEARPC_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
+	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@
 
 seaf_migrate_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
 

