--- a/tools/seafile-admin	2013-01-16 22:08:41.000000000 -0500
+++ b/tools/seafile-admin	2013-01-25 07:25:43.957175586 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/lib/seafile/seafileenv/bin/python2
 # coding: UTF-8
 
 '''This is the helper script to setup/manage your seafile server
@@ -642,7 +642,8 @@
 
 def start_seahub_gunicorn():
     argv = [
-        'gunicorn_django',
+        PYTHON,
+        '/usr/bin/gunicorn_django-python2',
         '-c', conf[CONF_SEAHUB_CONF],
         '-b', '0.0.0.0:%s' % conf[CONF_SEAHUB_PORT],
     ]
@@ -871,4 +872,4 @@
     args.func(args)
 
 if __name__ == '__main__':
-    main()
\ No newline at end of file
+    main()
