--- /etc/pam.d/su-l	2015-01-03 21:57:53.688740831 -0500
+++ su-l	2015-01-03 22:04:42.892089809 -0500
@@ -1,9 +1,13 @@
 #%PAM-1.0
+auth            sufficient      pam_ldap.so
 auth		sufficient	pam_rootok.so
 # Uncomment the following line to implicitly trust users in the "wheel" group.
 #auth		sufficient	pam_wheel.so trust use_uid
 # Uncomment the following line to require a user to be in the "wheel" group.
 #auth		required	pam_wheel.so use_uid
-auth		required	pam_unix.so
+auth		required	pam_unix.so use_first_pass
+account         sufficient      pam_ldap.so
 account		required	pam_unix.so
+session         required        pam_mkhomedir.so skel=/etc/skel umask=0022
+session         sufficient      pam_ldap.so
 session		required	pam_unix.so
